### PR TITLE
style: Improve output

### PR DESCRIPTION
**Pull request by gibson042**
* Update error descriptions for consistency and detail
* Show usage when required arguments are missing
* Send errors to stderr
* Set exit status to nonzero when there is an error

Also includes some go imports cleanup and better use of [`time`](https://pkg.go.dev/time) functions.

### CODE REVIEW

File: estimator/estimator.go [hunk 1]
```diff
@@ -154,7 +154,7 @@ func (e *Estimator) getAvgBlockDurationNanos() (time.Duration, error) {
 
 func (e *Estimator) CalcBlock(ttime time.Time) (int64, error) {
 	if ttime.Unix() <= time.Now().Unix() {
-		return 0, fmt.Errorf("time to estimate is < less than the current time")
+		return 0, fmt.Errorf("date to estimate must be in the future")
```
Comment (dckc) [inline estimator/estimator.go]: should that be "height to estimate ..."?

the user doesn't give a date as input, do they?

File: main.go [hunk 2]
```diff
@@ -11,30 +12,31 @@ import (
 	"github.com/araddon/dateparse"
 )
 
-func main() {
-
+func run() (exitStatus int, err error) {
 	fHeight := flag.Int64("height", 0, "specific height to estimate the time of")
 	fDate := flag.String("date", "", "specific time to estimate height at, e.g. Mon Jan 2 15:04:05 MST 2006")
-	fSamples := flag.Int64("samples", 100, "number of samples to take")
+	fSamples := flag.Int64("samples", 100, "count of samples to take")
```
Comment (dckc) [inline main.go:18]: "number of samples" is idiomatic, IME. count is an observation, not an input.
(not critical)

File: main.go [hunk 3]
```diff
@@ -11,30 +12,31 @@ import (
...
 	if *fHeight > 0 && *fDate != "" {
-		fmt.Println("must specify either height or date estimation")
-		return
+		return 64, fmt.Errorf("must not specify both height and date")
 	}
 
 	if *fSamples <= 0 {
-		fmt.Println("samples must be positive")
-		return
+		return 1, fmt.Errorf("sample count must be positive")
```
Comment (dckc) [inline main.go:31]: hm. this does argue for "count" over "number", I suppose

File: main.go [hunk 4]
```diff
@@ -11,30 +12,31 @@ import (
...
 	fThreads := flag.Int64("threads", 6, "number of threads to use")
-	fStatmode := flag.String("statmode", "mean", "statmode: mean or median")
+	fStatmode := flag.String("statmode", "mean", `how to estimate from past blocks, "mean" or "median"`)
 	fTimezones := flag.String("timezones", "Local,UTC,Asia/Tokyo,Australia/NSW,Asia/Istanbul,US/Pacific,US/Eastern", "comma-separated list of timezones")
 	flag.Parse()
 
 	if *fHeight > 0 && *fDate != "" {
-		fmt.Println("must specify either height or date estimation")
-		return
+		return 64, fmt.Errorf("must not specify both height and date")
```
Comment (dckc) [inline main.go:27]: :clap: for errors to stderr

File: main.go [hunk 5]
```diff
@@ -43,57 +45,66 @@ func main() {
...
-				fmt.Printf("%"+fmt.Sprintf("%d", ml)+"s: %s\n", tz, estTime.In(loc).Format(time.UnixDate))
+				fmt.Printf("%*s: %s\n", width, tz, estTime.In(loc).Format(time.UnixDate))
 			}
 		}
 	} else if *fDate != "" {
 		date, err := dateparse.ParseLocal(*fDate)
 		if err != nil {
-			fmt.Println(err.Error())
-			return
+			return 1, err
```
Comment (dckc) [inline main.go:76]: I wonder how much diagnostic info this loses.

**Review by dckc — APPROVED** [review]
Looks OK.

This isn't a completely thorough review. I'm not practiced at reviewing go code, and I started to skim a bit. So you might want more eyes on it.

File: main.go (hunk 5 above)
Comment (gibson042) [reply main.go:76]: None, because the error is still printed to stderr.

**Pull request merged**